### PR TITLE
Add delete functionality for samples

### PR DESCRIPTION
**Pull request by zteezy19**
Added delete functionality for issue #67. Used react-icons for trash icon for delete button. delete button is placed next to edit button.
Used the /api/v1/sqldelete/{id} API endpoint to perform the deletion.
Also added a confirmation prompt before deletion to prevent accidental removals using the confirmDialog function as well as some additional error handling.

### CODE REVIEW

**Review by sethumadhavan-k — CHANGES_REQUESTED** [review]

**Comment by zteezy19** [comment]
I've made the requested changes, thanks for the feedback. Please review it when you can @sethumadhavan-k

**Review by sethumadhavan-k — CHANGES_REQUESTED** [review]

**Comment by zteezy19** [comment]
Hi @sethumadhavan-k, I have adjusted as instructed. Sorry for the trouble

**Comment by sethumadhavan-k** [comment]
@zteezy19  let me check

**Review by sethumadhavan-k — CHANGES_REQUESTED** [review]

**Comment by zteezy19** [comment]
Hi @sethumadhavan-k, I've changed the name of the confirmDailog function to confirmDialog to fix the typo, and also adjusted the usage of the function in the Samples.jsx. Please take a look thanks

**Comment by sethumadhavan-k** [comment]
@zteezy19 The parameter of the confirmDialog function is not an object. Please ensure it follows the example provided, and for now, don't change the function name.

example :
confirmDailog("Are you sure you want to delete this?", "", {deleteicon}, {cancelicon}, {confirmButtonText}, {deleteButtononClickFunction})

**Comment by zteezy19** [comment]
Hi @sethumadhavan-k , I think I've adjusted as requested.

**Comment by sethumadhavan-k** [comment]
@zteezy19  let me check this.

**Comment by sethumadhavan-k** [comment]
@zteezy19 getting  this error in frontend 

Screenshot 2024-10-15 at 6 51 04 PM

File: ui/src/services/Sample.js [hunk 1]
```diff
@@ -1,25 +1,28 @@
...
+    sql_metadata: {
+      query: data.query,
+      metadata: data.metadata
+    }
+  })
 }
 
-
+export const deleteSample = (sampleId) => {
+  return DeleteService(`${API_URL}/sql/delete/${sampleId}`);
```
Comment (sethumadhavan-k) [inline ui/src/services/Sample.js:27]: delelet API url is  sqldelete/{id} not sql/delete/{id}

**Review by sethumadhavan-k — CHANGES_REQUESTED** [review]

**Comment by zteezy19** [comment]
Hi, I've fixed the API url typo.

**Comment by sethumadhavan-k** [comment]
@zteezy19 let check

**Comment by sethumadhavan-k** [comment]
@zteezy19 please update the sample delete  API url  `/sqldelete/{id}` also i am getting the same error 

Screenshot 2024-10-15 at 6 51 04 PM

**Comment by zteezy19** [comment]
Hi, sorry, I did not commit the codes properly. The changes should be committed. so sorry for the newbie mistakes

**Comment by sethumadhavan-k** [comment]
@zteezy19 please join on our slack channel https://join.slack.com/t/theailounge/shared_invite/zt-2ogkrruyf-FPOHuPr5hdqXl34bDWjHjw, i will assist there

**Pull request closed without merging**